applications/mne_scan/plugins/eegosports/FormFiles/eegosportsimpedancewidget.h

```diff
@@ -0,0 +1,185 @@
+
+namespace EEGOSPORTSPLUGIN
+{
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// USED NAMESPACES
+//=============================================================================================================
+
+using namespace Eigen;
+using namespace UTILSLIB;
+using namespace DISPLIB;
+
+//*************************************************************************************************************
+//=============================================================================================================
+// FORWARD DECLARATIONS
+//=============================================================================================================
+
+class EEGoSports;
+
+//=============================================================================================================
+/**
+* DECLARE CLASS EEGoSportsImpedanceWidget
+*
+* @brief The EEGoSportsImpedanceWidget class provides the EEGoSportsImpedanceWidget configuration window.
+*/
+class EEGoSportsImpedanceWidget : public QWidget
+{
+    Q_OBJECT
+
+public:
+    explicit EEGoSportsImpedanceWidget(EEGoSports* pEEGoSports, QWidget *parent = 0);
+    ~EEGoSportsImpedanceWidget();
+
+    //=========================================================================================================
+    /**
+    * Updates the values of the electrodes placed in the QGraphicsScene.
+    */
+    void updateGraphicScene(VectorXd matValue);
+
+    //=========================================================================================================
+    /**
+    * Initialises the 2D positions of the electrodes in the QGraphicsScene.
+    */
+    void initGraphicScene();
+
+private:
+    EEGoSports*                                       m_pEEGoSports;                    /**< The pointer back to the EEGoSports plugin.*/
+
+    EEGoSportsImpedanceScene*                         m_qGScene;                  /**< The QGraphicScene.*/
+
+    QMap< QString, int >                        m_qmElectrodeNameIndex;     /**< Lookup table for electrode name and their corresponding index in the received data matrix.*/
+
+    Ui::EEGoSportsImpedanceWidget*                    ui;                         /**< The user interface for the EEGoSportsImpedanceWidget.*/
+
+    QSharedPointer<ColorMap>                    m_cbColorMap;               /**< The pointer the colormap object.*/
+
+    double                                      m_dMaxImpedance;            /**< Maximum impedance value. This is a fixed value to scale the color map.*/
+
+    //=========================================================================================================
+    /**
+    * Adds an electrode item to the QGraphicScene.
+    */
+    void addElectrodeItem(QString electrodeName, QVector2D position);
```

Comment:
add @param[in] doxygen docu. Also, make sure to use const reference whenever possible.